UI Bug Fixes

kylehickinson opened the pull request:
Resolves #335, #338, #339, #340, #341, #344

## Pull Request Checklist

- [x] My patch has a standard commit message that looks like `Fix #123: This fixes the shattered coffee cup!`
- [ ] I have updated the *Unit Tests* to cover new or changed functionality
- [ ] I have updated the *UI Tests* to cover new or changed functionality
- [ ] I have made sure that localizable strings use `NSLocalizableString()`

## Screenshots

_None included_

## Notes for testing this patch

_None included_

kylehickinson committed:
Fix #339: Tab tray toolbar respects now left & right edges safe area

kylehickinson committed:
Fix #341: Back/forward history list now respects left & right safe area

kylehickinson committed:
Fix #340: Menu button in URL bar now respects left safe area when active

kylehickinson committed:
Fix #344: Favourites list now respects left & right safe area

kylehickinson committed:
Fix #338: Favourites cell size now resizes from landscape to portrait

kylehickinson committed:
Fix #335: Track current tab count so it doesn't fall out of sync

kylehickinson commented:
Please "Rebase and merge" when merging so it applies each commit individually

iccub reviewed (changes requested):
Code looks good but found one regression, on safe area iPhones in landscape in private mode.

<img width="950" alt="zrzut ekranu 2018-10-19 o 14 55 55" src="https://user-images.githubusercontent.com/6950387/47219492-3b27c600-d3af-11e8-9818-f77248353793.png">

It looks good on latest development.

kylehickinson commented:
Fixed! Also fixed so it does not cutoff favourites that fall under the home bar

![simulator screen shot - iphone x - 2018-10-19 at 09 16 31](https://user-images.githubusercontent.com/529104/47220396-ad80b200-d37f-11e8-9c94-e27943f7725b.png)
